src/identity.rs

```diff
@@ -221,13 +221,19 @@ impl Password {
         S2: Into<String>,
         S3: Into<String>,
     {
-        let url = auth_url.into_url()?;
-        // TODO: more robust logic?
-        let token_endpoint = if url.path().ends_with("/v3") {
+        let mut url = auth_url.into_url()?;
```

Comment:
Note: `url` is a local variable, mutating it is totally fine.

Reply:
Are you not concerned that for the purpose of building the `tokens` URL, we are changing the base auth_url (stripping the trailing slash from `Password::auth_url()`)?

cf:
https://github.com/dtantsur/rust-osauth/pull/14/files#diff-3d59bc47b6ad2d1cb1349abf34de2280R620

Reply:
Nope. Actually, one of the cool things about Rust is that it makes tracking such things easier. `IntoUrl.into_url` returns a `Url`, not a `&mut Url`, so we're mutating a local variable.